pennylane/qchem/integrals.py

```diff
@@ -304,13 +313,16 @@ def _overlap_integral(*args):
         Returns:
             array[float]: the overlap integral between two contracted Gaussian orbitals
         """
-
         args_a = [arg[0] for arg in args]
         args_b = [arg[1] for arg in args]
-        alpha, ca, ra = _generate_params(basis_a.params, args_a)
-        beta, cb, rb = _generate_params(basis_b.params, args_b)
-
-        if basis_a.params[1].requires_grad or normalize:
+        alpha, ca, ra = _generate_params(basis_a.params, args_a, argnum)
+        beta, cb, rb = _generate_params(basis_b.params, args_b, argnum)
+
+        if (
+            getattr(basis_a.params[1], "requires_grad", False)
+            or normalize
+            or (argnum is not None and 1 in argnum)
```

Comment:
Let's try to put `argnum` outside for these functions (here and at other ones) first to handle `None-Int-Iterable` cases and then use a simplified conditional check here.